src/hotspot/share/classfile/classLoaderData.hpp

```diff
@@ -202,10 +202,12 @@ class ClassLoaderData : public CHeapObj<mtClass> {
   // The "claim" is typically used to check if oops_do needs to be applied on
   // the CLD or not. Most GCs only perform strong marking during the marking phase.
   enum Claim {
-    _claim_none         = 0,
-    _claim_finalizable  = 2,
-    _claim_strong       = 3,
-    _claim_other        = 4
+    _claim_none                    = 0,
+    _claim_finalizable             = 2,
+    _claim_strong                  = 3,
+    _claim_other                   = 4,
+    _claim_strong_g1_fullgc_mark   = 8,
+    _claim_strong_g1_fullgc_adjust = 16
```

Comment:
I do agree that the naming is not optimal and I wonder if using more generic names would be better for now. Like just `_claim_mark` and `_claim_adjust` and look at making them GC-specific should be a different change that addresses all GC specific values. 

I also think `_claim_other` could continue to be last.